src/modules/songs/components/SongDetailsPage/SongDetailsPageHeader/SongDetailsPageHeader.tsx

```
@@ -30,8 +67,13 @@ export const SongDetailsPageHeader: React.FC<SongDetailsPageHeaderProps> = ({
         setIsEditing={setIsEditingName}
       />
       <HStack className="items-start gap-2">
-        <Button variant="outline" className="hidden md:flex">
-          <Heart />
+        <Button
+          variant="outline"
+          className="hidden md:flex"
+          onClick={updateSongFavoriteStatus}
+          disabled={updateSongFavoriteStatusMutation.isPending}
+        >
+          <Heart weight={!!song.favoritedAt ? "fill" : "regular"} />
```

Biome (1.9.4)

[error] Avoid redundant double-negation.

It is not necessary to use double-negation when a value will already be coerced to a boolean.
Unsafe fix: Remove redundant double-negation


(lint/complexity/noExtraBooleanCast)